# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jsapi.cpp
# Commit: 5c5cf716aa0b
# Full Hash: 5c5cf716aa0b01aecd8702254427825c9de1b8fa
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2023-06-08 03:42:08
# Regressor Bug: 1835876
# File Overlap Count: 1
# Description:
#   Bug 1835876 part 2 - Disable code write protection in content processes. r=nbp
#   
#   We've worked with the security and performance teams to re-evaluate the W^X policy
#   we have in place to mark JIT code memory pages as either writable or executable
#   (but not both).
# ==============================================================================

diff -r 028f981600d7 -r 5c5cf716aa0b js/src/jsapi.cpp
--- a/js/src/jsapi.cpp	Wed Jun 07 16:34:51 2023 +0000
+++ b/js/src/jsapi.cpp	Wed Jun 07 16:34:51 2023 +0000
@@ -4225,6 +4225,9 @@
     case JSJITCOMPILER_SPECTRE_JIT_TO_CXX_CALLS:
       jit::JitOptions.spectreJitToCxxCalls = !!value;
       break;
+    case JSJITCOMPILER_WRITE_PROTECT_CODE:
+      jit::JitOptions.maybeSetWriteProtectCode(!!value);
+      break;
     case JSJITCOMPILER_WATCHTOWER_MEGAMORPHIC:
       jit::JitOptions.enableWatchtowerMegamorphic = !!value;
       break;
@@ -4314,6 +4317,9 @@
     case JSJITCOMPILER_SPECTRE_JIT_TO_CXX_CALLS:
       *valueOut = jit::JitOptions.spectreJitToCxxCalls ? 1 : 0;
       break;
+    case JSJITCOMPILER_WRITE_PROTECT_CODE:
+      *valueOut = jit::JitOptions.writeProtectCode ? 1 : 0;
+      break;
     case JSJITCOMPILER_WATCHTOWER_MEGAMORPHIC:
       *valueOut = jit::JitOptions.enableWatchtowerMegamorphic ? 1 : 0;
       break;